crates/hyperswitch_connectors/src/connectors/novalnet/transformers.rs

```diff
@@ -1064,10 +1085,90 @@ impl<F>
 }
 
 //TODO: Fill the struct with respective fields
-#[derive(Default, Debug, Serialize, Deserialize, PartialEq)]
+#[derive(Default, Debug, Serialize, Deserialize)]
 pub struct NovalnetErrorResponse {
-    pub status_code: u16,
+    pub status_code: u32,
     pub code: String,
     pub message: String,
     pub reason: Option<String>,
 }
+
+#[derive(Display, Debug, Serialize, Deserialize)]
+#[strum(serialize_all = "SCREAMING_SNAKE_CASE")]
+#[serde(rename_all = "SCREAMING_SNAKE_CASE")]
+pub enum WebhookEventType {
+    Payment,
+    TransactionCapture,
+    TransactionCancel,
+    TransactionRefund,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct NovalnetWebhookEvent {
+    pub checksum: String,
+    pub tid: i64,
+    pub parent_tid: Option<i64>,
+    #[serde(rename = "type")]
+    pub event_type: WebhookEventType,
+}
+
+#[derive(Debug, Serialize, Deserialize, Clone)]
+#[serde(untagged)]
+pub enum NovalnetWebhookTransactionData {
+    OtherTransactionData(NovalnetSyncResponseTransactionData),
+    CaptureTransactionData(NovalnetCaptureTransactionData),
+    CancelTransactionData(NovalnetPaymentsResponseTransactionData),
+    RefundsTransactionData(NovalnetRefundsTransactionData),
+}
+#[derive(Serialize, Deserialize, Debug)]
+pub struct NovalnetWebhookNotificationResponse {
+    pub customer: Option<NovalnetResponseCustomer>,
+    pub event: NovalnetWebhookEvent,
+    pub merchant: NovalnetResponseMerchant,
+    pub result: ResultData,
+    pub transaction: NovalnetWebhookTransactionData,
+}
+
+pub fn is_refund_event(event_code: &WebhookEventType) -> bool {
+    matches!(event_code, WebhookEventType::TransactionRefund)
+}
+
+pub fn get_incoming_webhook_event(
+    status: WebhookEventType,
+    transaction_status: NovalnetTransactionStatus,
+) -> IncomingWebhookEvent {
+    match status {
+        WebhookEventType::Payment => match transaction_status {
+            NovalnetTransactionStatus::Confirmed | NovalnetTransactionStatus::Success => {
+                IncomingWebhookEvent::PaymentIntentSuccess
+            }
+            NovalnetTransactionStatus::OnHold => {
+                IncomingWebhookEvent::PaymentIntentAuthorizationSuccess
+            }
+            NovalnetTransactionStatus::Pending | NovalnetTransactionStatus::Progress => {
```

Comment:
We get progress incase of authentication pending in payments response, are you sure you want to update this to Processing ?